### PR TITLE
Selectively install thriftpy2 requirement based on python version

### PR DESCRIPTION
**Pull request by dknupp**
This patch addresses both #329 and #326.

Tested by setting up two different virtualenvs, and confimring that
the expected requirements were installed, and that the connect function
could be imported from impala.dbapi.

**Python 2**
```
[dknupp@mbp ~/Documents/code]$ virtualenv /tmp/python2_venv
New python executable in /tmp/python2_venv/bin/python
Installing setuptools, pip, wheel...
done.
[dknupp@mbp ~/Documents/code]$ source /tmp/python2_venv/bin/activate
(python2_venv) [dknupp@mbp ~/Documents/code]$ python --version
Python 2.7.10
(python2_venv) [dknupp@mbp ~/Documents/code]$ pip list
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Package    Version
---------- -------
pip        19.0.3
setuptools 40.8.0
wheel      0.33.1
(python2_venv) [dknupp@mbp ~/Documents/code]$ pip install -e impyla/
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Obtaining file:///Users/dknupp/Documents/code/impyla
Collecting six (from impyla==0.14.2.1+8.gc010d59)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting bitarray (from impyla==0.14.2.1+8.gc010d59)
Collecting thrift>=0.9.3 (from impyla==0.14.2.1+8.gc010d59)
Installing collected packages: six, bitarray, thrift, impyla
  Running setup.py develop for impyla
Successfully installed bitarray-0.8.3 impyla six-1.12.0 thrift-0.11.0
(python2_venv) [dknupp@mbp ~/Documents/code]$ pip list
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Package    Version             Location
---------- ------------------- -----------------------------------
bitarray   0.8.3
impyla     0.14.2.1+8.gc010d59 /Users/dknupp/Documents/code/impyla
pip        19.0.3
setuptools 40.8.0
six        1.12.0
thrift     0.11.0
wheel      0.33.1
(python2_venv) [dknupp@mbp ~/Documents/code]$ python
Python 2.7.10 (default, Aug 17 2018, 19:45:58)
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from impala.dbapi import connect
>>> ^D
```

**Python 3**
```
[dknupp@mbp ~/Documents/code]$ virtualenv --python /usr/local/bin/python3 /tmp/python3_venv
Running virtualenv with interpreter /usr/local/bin/python3
Using base prefix '/usr/local/Cellar/python/3.7.2_1/Frameworks/Python.framework/Versions/3.7'
New python executable in /tmp/python3_venv/bin/python3.7
Also creating executable in /tmp/python3_venv/bin/python
Installing setuptools, pip, wheel...
done.
[dknupp@mbp ~/Documents/code]$ source /tmp/python3_venv/bin/activate
(python3_venv) [dknupp@mbp ~/Documents/code]$ python --version
Python 3.7.2
(python3_venv) [dknupp@mbp ~/Documents/code]$ pip list
Package    Version
---------- -------
pip        19.0.3
setuptools 40.8.0
wheel      0.33.1
(python3_venv) [dknupp@mbp ~/Documents/code]$ pip install -e impyla/
Obtaining file:///Users/dknupp/Documents/code/impyla
Collecting six (from impyla==0.14.2.1+8.gc010d59)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting bitarray (from impyla==0.14.2.1+8.gc010d59)
Collecting thrift>=0.9.3 (from impyla==0.14.2.1+8.gc010d59)
Collecting thriftpy2==0.4.0 (from impyla==0.14.2.1+8.gc010d59)
Collecting ply<4.0,>=3.4 (from thriftpy2==0.4.0->impyla==0.14.2.1+8.gc010d59)
  Using cached https://files.pythonhosted.org/packages/a3/58/35da89ee790598a0700ea49b2a66594140f44dec458c07e8e3d4979137fc/ply-3.11-py2.py3-none-any.whl
Installing collected packages: six, bitarray, thrift, ply, thriftpy2, impyla
  Running setup.py develop for impyla
Successfully installed bitarray-0.8.3 impyla ply-3.11 six-1.12.0 thrift-0.11.0 thriftpy2-0.4.0
(python3_venv) [dknupp@mbp ~/Documents/code]$ pip list
Package    Version             Location
---------- ------------------- -----------------------------------
bitarray   0.8.3
impyla     0.14.2.1+8.gc010d59 /Users/dknupp/Documents/code/impyla
pip        19.0.3
ply        3.11
setuptools 40.8.0
six        1.12.0
thrift     0.11.0
thriftpy2  0.4.0
wheel      0.33.1
(python3_venv) [dknupp@mbp ~/Documents/code]$ python
Python 3.7.2 (default, Jan 13 2019, 12:50:01)
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from impala.dbapi import connect
>>>
```

### CODE REVIEW

File: jenkins/run-dbapi.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ conda info -a
 CONDA_ENV_NAME=pyenv-impyla-dbapi-test
 conda create -y -q -n $CONDA_ENV_NAME python=$PYTHON_VERSION pip
 source activate $CONDA_ENV_NAME
-pip install thriftpy sqlalchemy
+pip install thriftpy2 sqlalchemy
```
Comment (lekv) [inline jenkins/run-dbapi.sh]: Isn't this going to be an issue with python 2, i.e. we're unconditionally installing thriftpy2 here.

Comment (dknupp) [reply jenkins/run-dbapi.sh]: thriftpy2 is compatible with python2.x and 3.x, and is considered a replacement for thriftpy, which is deprecated:
https://github.com/Thriftpy/thriftpy2

Comment (lekv) [reply jenkins/run-dbapi.sh]: Thanks for the clarification.

**Comment by ethe** [comment]
Please do not block the version of thriftpy2, make the version greater than 0.4.0 and less than 0.5.0 is fine.

**Pull request merged**